sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -48,3 +48,24 @@
 #else
 #define __SYCL_INLINE_CONSTEXPR static constexpr
 #endif
+
+#ifndef __SYCL_HAS_CPP_ATTRIBUTE
+#if defined(__cplusplus) && defined(__has_cpp_attribute)
+#define __SYCL_HAS_CPP_ATTRIBUTE(x) __has_cpp_attribute(x)
+#else
+#define __SYCL_HAS_CPP_ATTRIBUTE(x) 0
+#endif
+#endif
+
+#if defined(__cplusplus) && __cplusplus > 201402L &&                           \
+    __SYCL_HAS_CPP_ATTRIBUTE(fallthrough)
+#define __SYCL_FALLTHROUGH [[fallthrough]]
+#elif __SYCL_HAS_CPP_ATTRIBUTE(gnu::fallthrough)
+#define __SYCL_FALLTHROUGH [[gnu::fallthrough]]
+#elif __has_attribute(fallthrough)
+#define __SYCL_FALLTHROUGH __attribute__((fallthrough))
+#elif __SYCL_HAS_CPP_ATTRIBUTE(clang::fallthrough)
+#define __SYCL_FALLTHROUGH [[clang::fallthrough]]
+#else
+#define __SYCL_FALLTHROUGH
+#endif
```

Comment:
late nit:
```suggestion
#endif
#endif // __SYCL_FALLTHROUGH
```

Reply:
Fixed in https://github.com/intel/llvm/pull/2980/commits/a60e63f1c597cd83e588fe0cf5337929998ba83f